ext/reflection/php_reflection.c

```diff
@@ -2125,7 +2125,137 @@ ZEND_METHOD(ReflectionFunctionAbstract, getNumberOfRequiredParameters)
 }
 /* }}} */
 
-/* {{{ Returns an array of parameter objects for this function */
+/* {{{ Returns whether a parameter exists or not */
+ZEND_METHOD(ReflectionFunctionAbstract, hasParameter)
+{
+    reflection_object *intern;
+    zend_function *fptr;
+    struct _zend_arg_info *arg_info;
+    zval *parameter;
+    uint32_t i, num_args, position;
+
+    if (zend_parse_parameters(ZEND_NUM_ARGS(), "z", &parameter) == FAILURE) {
+        RETURN_THROWS();
+    }
+
+    GET_REFLECTION_OBJECT_PTR(fptr);
+
+    num_args = fptr->common.num_args;
+    arg_info= fptr->common.arg_info;
+
+    if (fptr->common.fn_flags & ZEND_ACC_VARIADIC) {
+        num_args++;
+    }
+
+    if (!num_args) {
+        RETURN_THROWS();
+    }
+
+    switch(Z_TYPE_P(parameter)) {
+        case IS_LONG:
```

Comment:
I intentionally wrote "number" or numeric string because you should respect the type juggling rules in the weak mode: so if I pass in "1" then the method should interpret it as a positional argument number.